packages/dds/tree/src/feature-libraries/treeTextFormat.ts

```diff
@@ -0,0 +1,225 @@
+ * summaries should be able to reference unloaded chunks instead of having children directly.
+ * Leverage placeholders in the types below to accomplish this.
+ * Determine how this relates to Cursor: should cursor be generic over placeholder values?
+ * (Could use them for errors to allow non erroring cursors?)
+ *
+ * Note:
+ * Currently a lot of Tree's codebase is using json for serialization.
+ * Because putting json strings inside json works poorly (adds lots of escaping),
+ * for now this library actually outputs and inputs the Json compatible type PlaceholderTree
+ * rather than actual strings.
+ */
+
+/**
+ * Json compatible map as object.
+ * Keys are TraitLabels,
+ * Values are the content of the trait specified by the key.
+ * @public
+ */
+ export interface FieldMap<TChild> {
+    [key: string]: TreeNodeSequence<TChild>;
+}
+
+/**
+ * A sequence of Nodes that make up a trait under a Node
+ * @public
+ */
+export type TreeNodeSequence<TChild> = readonly TChild[];
+
+/**
+ * An object which may have traits with children of the given type underneath it
+ * @public
+ */
+export interface WithFields<TChild> {
+    fields?: Readonly<FieldMap<TChild>>;
+}
+
+/**
+ * The fields required by a node in a tree
+ * @public
+ */
+export interface NodeData {
+    /**
+     * A payload of arbitrary serializable data
+     */
+    value?: TreeValue;
+
+    /**
+     * The meaning of this node.
+     * Provides contexts/semantics for this node and its content.
+     * Typically use to associate a node with metadata (including a schema) and source code (types, behaviors, etc).
+     */
+    readonly type: TreeSchemaIdentifier;
+}
+
+/**
+ * Satisfies `NodeData` and may contain children under traits (which may or may not be `TreeNodes`)
+ * @public
+ */
+export interface TreeNode<TChild> extends NodeData, WithFields<TChild> {}
+
+/**
+ * A tree whose nodes are either TreeNodes or a placeholder
+ */
+export type PlaceholderTree<TPlaceholder = never> = TreeNode<PlaceholderTree<TPlaceholder>> | TPlaceholder;
+
+/**
+ * An ITreeCursor implementation for PlaceholderTree.
+ *
+ * TODO: object-forest's cursor is mostly a superset of this functionality.
+ * Maybe do a refactoring to deduplicate this.
+ */
+export class TextCursor implements ITreeCursor {
+    // Ancestors traversed to visit this node (including this node).
+    private readonly parentStack: PlaceholderTree[] = [];
+    // Keys traversed to visit this node
+    private readonly keyStack: FieldKey[] = [];
+    // Indices traversed to visit this node
+    private readonly indexStack: number[] = [];
+
+    private siblings: readonly PlaceholderTree[];
+    private readonly root: readonly PlaceholderTree[];
+
+    public constructor(root: PlaceholderTree) {
+        this.root = [root];
+        this.indexStack.push(0);
+        this.siblings = this.root;
+        this.parentStack.push(this.siblings[0]);
```

Comment:
Why use `this.siblings[0]` instead of `root`?

Reply:
Thats a good change. I'll make it in my next PR.